tests/ImageSharp.Tests/Processing/Transforms/TransformTests.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace SixLabors.ImageSharp.Tests.Processing.Transforms
+{
+    using System.Numerics;
+    using System.Reflection;
+
+    using SixLabors.ImageSharp.PixelFormats;
+    using SixLabors.ImageSharp.Processing;
+    using SixLabors.Primitives;
+
+    using Xunit;
+
+    public class TransformTests : FileTestBase
+    {
+        public static readonly TheoryData<float, float, float> TransformValues
+            = new TheoryData<float, float, float>
+                  {
+                      { 20, 10, 45 },
+                      { -20, -10, 45 }
+                  };
+
+        public static readonly List<string> ResamplerNames
+            = new List<string>
+                  {
+                      nameof(KnownResamplers.Bicubic),
+                      //nameof(KnownResamplers.Box),
+                      //nameof(KnownResamplers.CatmullRom),
+                      //nameof(KnownResamplers.Hermite),
+                      //nameof(KnownResamplers.Lanczos2),
+                      //nameof(KnownResamplers.Lanczos3),
+                      //nameof(KnownResamplers.Lanczos5),
+                      //nameof(KnownResamplers.Lanczos8),
+                      //nameof(KnownResamplers.MitchellNetravali),
+                      //nameof(KnownResamplers.NearestNeighbor),
+                      //nameof(KnownResamplers.Robidoux),
+                      //nameof(KnownResamplers.RobidouxSharp),
+                      //nameof(KnownResamplers.Spline),
+                      //nameof(KnownResamplers.Triangle),
+                      //nameof(KnownResamplers.Welch),
+                  };
+
+        [Theory]
+        [WithFileCollection(nameof(DefaultFiles), nameof(TransformValues), DefaultPixelType)]
```

Comment:
Let's do this with the test pattern instead! (-> smaller actual and reference outputs).

I also suggest to have two `[Theory]`-s to cover this stuff:
- One for going through all the ResamplerName-s using it as a theory parameter + fixing a single transform value
- Another one to test different transform parameters with a single resampler.

This way we can cover everything without bloating the test-space.

Reply:
I will do this very soon!